### PR TITLE
Fix coco style bounding box annotations

### PR DESCRIPTION
**Pull request by ramonhollands**

### CODE REVIEW

**Comment by jackdaw213** [comment]
@ramonhollands Just tried out the pull request, it works but AP can't exceed 1
image
The mock dataset is better 
image
but not as good as using segmentation 
image

File: yolo/utils/dataset_utils.py [hunk 1]
```diff
@@ -104,7 +104,14 @@ def scale_segmentation(
         if "segmentation" in anno:
             seg_list = [item for sublist in anno["segmentation"] for item in sublist]
         elif "bbox" in anno:
-            seg_list = anno["bbox"]
+            x,y,width,height = anno["bbox"]
```
Comment (DmitriiSavin) [inline yolo/utils/dataset_utils.py:107]: Hi!
Thank you for your suggestions. I would just like to add 2 suggestions regarding your PR:
1. It might make more sense to mode this logic to [this](https://github.com/WongKinYiu/YOLO/blob/main/yolo/tools/data_loader.py#L126) function, as the current function is named `scale_segmentation` and supposed to do scaling, and not reformatting
2. It might be more transparent and convenient to use the already existing function from [here](https://github.com/WongKinYiu/YOLO/blob/main/yolo/utils/bounding_box_utils.py#L80): `transform_bbox`, to reformat the bboxes

All together the new function might look like this:
```
bbox = seg_data[1:]
bbox = transform_bbox(torch.tensor(bbox), "xywh -> xyxy")
bbox = torch.tensor([cls, *bbox])
bboxes.append(bbox)
```



it should look something like that:

Comment (ramonhollands) [reply yolo/utils/dataset_utils.py:107]: Hi @DmitriiSavin,

Thanks for thinking along! The problem of using the function 'load_valid_labels' is the fact that we don't know at that point whether we are dealing with segmentation or bbox annotations. We even might have a dataset containing both mixed. So I don't think this is a good spot.

'seg_data' might be segmentation data in your preview.

Or am I overlooking something?

Comment (DmitriiSavin) [reply yolo/utils/dataset_utils.py:107]: @ramonhollands Hi!

I see where you're coming from, and I had similar concerns. Here are a few points to consider:

1. In the `load_valid_labels` function, we currently use the variable names 'bbox' and 'bboxes' exclusively. If this function is intended to handle segmentation areas as well, we’d need to adjust the code to reflect that change.
2. The line `torch.tensor([cls, *valid_points.min(axis=0), *valid_points.max(axis=0)])` in `load_valid_labels` creates a tensor of length 5, which represents a bounding box rather than a segmentation area. This indicates that the function is designed to handle bounding box data, not segmentation data.

Regarding `seg_data`, you mentioned it might represent segmentation data. However, since we treat it as a bounding box in the current implementation, I’m not certain that `seg_data` could actually contain segmentation data.

Please feel free to let me know if I’ve misunderstood anything here, or if there's anything I may have missed!

Comment (ramonhollands) [reply yolo/utils/dataset_utils.py:107]: The code is indeed confusing. To my understanding the load_valid_labels function validates and converts segmentation annotations to bounding boxes. So we might want to rename that function or at least update the documentation to prevent confusion. 

The input to the load_valid_labels function is the output from this line: https://github.com/WongKinYiu/YOLO/blob/e78c98bfadd745e97070702023c39bc59ee9b6ac/yolo/tools/data_loader.py#L89

Comment (DmitriiSavin) [reply yolo/utils/dataset_utils.py:107]: > To my understanding the load_valid_labels function validates and converts segmentation annotations to bounding boxes
I'm uncertain if converting segmentation annotations to bounding boxes is a correct. The output of the `load_valid_labels` function consistently has a shape of `(number of annotations, 5)`:

```
        if bboxes:
            return torch.stack(bboxes)
        else:
            logger.warning("No valid BBox in {}", label_path)
            return torch.zeros((0, 5))
```

It seems to me that this code expects only bounding box annotations rather than segmentation areas as input. 
Let me know if I’m missing something, but I have some doubts that segmentation data will be handled correctly here.

Comment (henrytsui000) [reply yolo/utils/dataset_utils.py:107]: Handling segmentation data as input will be necessary in the future, as the Copy-and-Paste augmentation and segmentation task training will require this information. Currently, I’m only working with bounding boxes and converting segmentation data to bounding boxes. If possible, please help make the loader flexible enough to accommodate segmentation data in the future.

Comment (ramonhollands) [reply yolo/utils/dataset_utils.py:107]: I created a new PR
